src/services/offer.js

```diff
@@ -225,10 +227,15 @@ class OfferService {
         return constraints.length ? { "$and": constraints } : {};
     }
 
-    async getOfferById(offerId, user) {
-        const offer = await Offer.findById(offerId);
+    async getOfferById(offerId, user, showAdminReason = false) {
+        const offerQuery = Offer.findById(offerId);
+
+        if (!showAdminReason) offerQuery.select("-adminReason");
+
+        const offer = await offerQuery;
 
-        if (offer?.isHidden && !(user?.isAdmin || offer.owner.toString() === user?.company?._id.toString())) return null;
+        // this is a good fix for now but should be changed later
+        if (offer?.isHidden && !(user?.isAdmin || showAdminReason || offer.owner.toString() === user?.company?._id.toString())) return null;
```

Comment:
Why is the `showAdminReason` relevant in this line?

Reply:
If  `user?.isAdmin` was replaced with `showAdminReason`, some tests would fail because not all calls to `getOfferById` have a third parameter, thus, even though the user could be an admin, that information would never be used inside this method.
I can't think of a good solution off the top of my head, so any suggestions are welcome.

Reply:
What I meant was that `showAdminReason` has nothing to do with this line. It simply controls if the offer is to be returned or not, which should be refactored to use `req.hasAdminPermissions` instead of `user?.isAdmin`.

`showAdminReason` simply alters the query above, that's fine

Reply:
Oh ok, got it